src/plugins/discover/public/application/main/services/discover_state.ts

```diff
@@ -280,89 +251,40 @@ export function getDiscoverStateContainer({
     internalStateContainer.transitions.setSavedDataViews(dataViewList);
   };
 
+  const loadAndResolveDataView = async (id: string, actualSavedSearch: SavedSearch) => {
+    const nextDataViewData = await loadDataView(services.dataViews, services.uiSettings, id);
+    const nextDataView = resolveDataView(
+      nextDataViewData,
+      actualSavedSearch.searchSource,
+      services.toastNotifications
+    );
+    return { fallback: !nextDataViewData.stateValFound, dataView: nextDataView };
+  };
+
   return {
```

Comment:
At some point I think it would make sense to get rid of some of these methods that seem to just call the equivalent methods on `appStateContainer` since we expose `appState` directly anyway. For example, I'm not sure methods like `resetInitialAppState` and `getPreviousAppState` are necessary, and the duplication might just cause confusion. It might also solve the circular import issue between `discover_app_state_container` and `discover_state`.

Reply:
yes, this should happen soon, nice cleanup potential for sure!